environments/common/src/main/java/org/jboss/weld/environment/deployment/discovery/DiscoveryStrategyFactory.java

```diff
@@ -37,17 +37,22 @@ private DiscoveryStrategyFactory() {
      * @param resourceLoader
      * @param bootstrap
      * @param initialBeanDefiningAnnotations
+     * @param jandexStrategyDisabled
      * @return the discovery strategy
      */
     public static DiscoveryStrategy create(ResourceLoader resourceLoader, Bootstrap bootstrap,
-            Set<Class<? extends Annotation>> initialBeanDefiningAnnotations) {
-        if (Jandex.isJandexAvailable(resourceLoader)) {
-            CommonLogger.LOG.usingJandex();
-            try {
-                return Jandex.createJandexDiscoveryStrategy(resourceLoader, bootstrap, initialBeanDefiningAnnotations);
-            } catch (Exception e) {
-                throw CommonLogger.LOG.unableToInstantiate(Jandex.JANDEX_DISCOVERY_STRATEGY_CLASS_NAME,
+        Set<Class<? extends Annotation>> initialBeanDefiningAnnotations, boolean jandexStrategyDisabled) {
+        if (jandexStrategyDisabled) {
```

Comment:
Hmm, we could but I don't like it.
I specifically made it this way, so that we don't even need to look whether there is any Jandex on CP if this option is set.